langlib/lang.array/src/main/ballerina/src/lang.array/array.bal

```diff
@@ -48,50 +66,173 @@ public function iterator(Type[] arr) returns abstract object {
     return arrIterator;
 }
 
+# Returns a new array comprising of position and member pairs.
+#
+# + arr - the array
+# + return - array of position, member pair
 public function enumerate(Type[] arr) returns [int, Type][] = external;
 
-// Functional iteration
-
+# Returns a new array applying function `func` to each member of array `arr`.
+#
+# + arr - the array
+# + func - a function to apply to each member
+# + return - new array containing result of applying function `func` to each member
 public function 'map(Type[] arr, function(Type val) returns Type1 func) returns Type1[] = external;
 
+# Apply function `func` to each member of array `arr`.
+#
+# + arr - the array
+# + func - a function to apply to each member
 public function forEach(Type[] arr, function(Type val) returns () func) returns () = external;
 
+# Returns a new array constructed from those elements of 'arr' for which `func` returns true.
+#
+# + arr - the array
+# + func - a predicate to apply to each element to determine if it should be included
+# + return - new array only containig members which evaluate function 'func' to true
 public function filter(Type[] arr, function(Type val) returns boolean func) returns Type[] = external;
 
+# Reduce operate on each member of `arr` using combining function `func` to produce
```

Comment:
```suggestion
# Reduce operate on each member of the `arr` array using the `func` function to produce
```
What do we mean by "reduce operate"?

Reply:
'reduce' function operates on each member